recipe/gen_patch_json.py

```diff
@@ -692,11 +692,13 @@ def _gen_new_index(repodata, subdir):
         if any(dep.startswith("zstd >=1.4") for dep in deps):
             _pin_looser(fn, record, "zstd", max_pin="x.x")
 
-        # We pin MPI packages loosely so as to rely on their ABI compatibility                                    
+        # We pin MPI packages loosely so as to rely on their ABI compatibility
         if any(dep.startswith("openmpi >=4.0") for dep in deps):
             _pin_looser(fn, record, "openmpi", upper_bound="5.0")
         if any(dep.startswith("mpich >=3.3") for dep in deps):
             _pin_looser(fn, record, "mpich", upper_bound="4.0")
+        if any(dep.startswith("mpich >=3.4") for dep in deps):
```

Comment:
```suggestion
        if any(dep.startswith("mpich >=3.4,<3.5") for dep in deps):
```

Reply:
Why?

Reply:
Because repodata patching should be applied to older packages and not new ones.

Reply:
the current code generates this patch

```
linux-64::lalinference-2.0.6-mpi_mpich_py36h026e38f_0.tar.bz2
-    "mpich >=3.4.1,<3.5.0a0",
+    "mpich >=3.4.1,<4.0.0a0",
```

I think we want that patched too. your edit would exclude it.

Reply:
See my diff above.

Reply:
I pushed a change that gets both versions.